wasm/jit/engine.go

```diff
@@ -148,217 +401,52 @@ func (e *engine) Call(f *wasm.FunctionInstance, params ...uint64) (results []uin
 	// so we assign them in reverse order.
 	results = make([]uint64, len(f.FunctionType.Type.Results))
 	for i := range results {
-		results[len(results)-1-i] = e.pop()
+		results[len(results)-1-i] = e.popValue()
 	}
 	return
 }
 
-func (e *engine) Compile(f *wasm.FunctionInstance) error {
-	if f.IsHostFunction() {
-		e.compiledFunctions[f.Address] = &compiledFunction{
-			source:      f,
-			paramCount:  uint64(len(f.FunctionType.Type.Params)),
-			resultCount: uint64(len(f.FunctionType.Type.Results)),
-		}
-	} else {
-		cf, err := e.compileWasmFunction(f)
-		if err != nil {
-			return fmt.Errorf("failed to compile Wasm function: %w", err)
-		}
-		e.compiledFunctions[f.Address] = cf
-	}
-	return nil
-}
-
 func NewEngine() wasm.Engine {
 	return newEngine()
 }
 
-const initialStackSize = 1024
+// TODO: better make them configurable?
```

Comment:
should we? think about other JIT look at them and cite if they configure this! (when you get to the TODO)